### PR TITLE
Quitquitquit

### PR DESCRIPTION
**Pull request by Tomasz-Smelcerz-SAP**
**Description**
New Istio version 1.3.4 brings a standard way to terminate istio-pilot container with a dedicated HTTP endpoint: `/quitquitquit`.
We have to terminate this container in our tests, because when the test container finishes, the istio-pilot container still runs and the test infrastructure observes this as a never-ending run of the test scenario.
At the moment we're using brute-force approach: pkill of the istio-pilot process.
But this solution requires to use the flag `shareProcessNamespace: true`, which doesn't work in the Azure environment.

In order to run our tests in Azure, we have to switch to the `quitquitquit` approach.
Edit: It seems this flag is now working in AKS clusters, but we can encounter similar problems in other managed K8s environments, so the switch to `quitquitquit` is still important.

Changes proposed in this pull request:

- Switch to explicit termination of test pods using /quitquitquit istio-pilot endpoint.

*See also*
Implements: #4719

### CODE REVIEW

File: resources/assetstore/templates/tests/test.yaml [hunk 1]
```diff
@@ -29,8 +29,20 @@ spec:
       containers:
         - name: tests
           image: {{ .Values.global.containerRegistry.path }}/{{ .Values.global.asset_store_test.dir }}asset-store-test:{{ .Values.global.asset_store_test.version }}
-          command: ["/bin/sh"]
-          args: ["-c", "sleep 20; /app/main.test; exit_code=$?; pkill -INT pilot-agent; sleep 4; exit $exit_code;"]
+          command:
```
Comment (strekm) [inline resources/assetstore/templates/tests/test.yaml]: please remove `shareProcessNamespace`

File: resources/core/charts/kubeless/templates/tests/test.yaml [hunk 2]
```diff
@@ -21,8 +21,20 @@ spec:
       containers:
       - name: tests
         image: {{ .Values.global.containerRegistry.path }}/{{ .Values.global.kubeless_tests.dir }}kubeless-tests:{{ .Values.global.kubeless_tests.version }}
-        command: ["/bin/sh"]
-        args: ["-c", "sleep 20; /kubeless-tests; exit_code=$?; pkill -INT pilot-agent; sleep 4; exit $exit_code;"]
+        command:
```
Comment (strekm) [inline resources/core/charts/kubeless/templates/tests/test.yaml]: please remove `shareProcessNamespace`

File: resources/core/charts/console-backend-service/templates/tests/test.yaml [hunk 3]
```diff
@@ -68,7 +68,19 @@ spec:
           resources:
             limits:
               memory: 128Mi
-          command: ["/bin/sh"]
-          args: ["-c", "sleep 10; cd /app/tests/ && ./run-compiled-tests.sh; exit_code=$?; pkill -INT pilot-agent; sleep 4; exit $exit_code;"]
+          command:
```
Comment (strekm) [inline resources/core/charts/console-backend-service/templates/tests/test.yaml]: please remove `shareProcessNamespace`

File: tests/end-to-end/upgrade/chart/upgrade/templates/pod.yaml [hunk 4]
```diff
@@ -37,21 +38,22 @@ spec:
...
-        while true; do  
-        if pgrep -x "${SERVICE}"; then  
-          echo "---> ${SERVICE} is running."  
-            sleep 10  
-          else  
-            echo "---> ${SERVICE} has stopped, stopping istio"  
-            pkill -TERM pilot-agent 
-            break 
-          fi  
+        while true; do
```
Comment (strekm) [inline tests/end-to-end/upgrade/chart/upgrade/templates/pod.yaml]: please remove `shareProcessNamespace`

File: resources/assetstore/templates/tests/test.yaml (hunk 1 above)
Comment (Tomasz-Smelcerz-SAP) [reply resources/assetstore/templates/tests/test.yaml]: Done.

File: resources/core/charts/console-backend-service/templates/tests/test.yaml (hunk 3 above)
Comment (Tomasz-Smelcerz-SAP) [reply resources/core/charts/console-backend-service/templates/tests/test.yaml]: Done.

File: resources/core/charts/kubeless/templates/tests/test.yaml (hunk 2 above)
Comment (Tomasz-Smelcerz-SAP) [reply resources/core/charts/kubeless/templates/tests/test.yaml]: Done.

File: tests/end-to-end/upgrade/chart/upgrade/templates/pod.yaml (hunk 4 above)
Comment (Tomasz-Smelcerz-SAP) [reply tests/end-to-end/upgrade/chart/upgrade/templates/pod.yaml]: Done.

**Pull request merged**